File: Python/3105. Longest Strictly Increasing or Strictly Decreasing Subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        result = 0
        order = 0
        temp = 1

        for i in range(1, len(nums)):
            if order != -1 and nums[i-1] < nums[i]:
                order =  1
                temp += 1
            elif order != 1 and nums[i-1] > nums[i]:
                order = -1
                temp += 1
            else:
                order *= -1
                result = max(result, temp)
                temp = 1 + int(nums[i-1] != nums[i])

        result = max(result, temp)

        return result

File: C/3105. Longest Strictly Increasing or Strictly Decreasing Subarray.c
int max(int a, int b) {
    return (a > b) ? a : b;
}

int longestMonotonicSubarray(int* nums, int numsSize) {
    int order = 0, result = 0, temp = 1;

    for (int* i = nums + 1; i < nums + numsSize; i++) {
        if (order != -1 && *(i - 1) < *(i)) {
            order = 1;
        } else if (order != 1 && *(i - 1) > *(i)) {
            order = -1;
        } else {
            order *= -1;
            result = max(result, temp);
            temp = *(i - 1) == *(i) ? 0 : 1;
        }
        temp++;
    }

    return max(result, temp);
}
